src/protocols/secure_channel/CASESession.cpp

```diff
@@ -2690,4 +2714,23 @@ SessionEstablishmentStage CASESession::MapCASEStateToSessionEstablishmentStage(S
     }
 }
 
+#if CHIP_WITH_NLFAULTINJECTION
+
+// safely corrupt the ephemeral key by serializing it, modifying the serialized buffer, and deserializing into a new keypair.
+// This avoids using const_cast which could be unsafe and is guaranteed to work across all keystore implementations.
+void CASESession::TestOnlyInjectFaultIntoInitiatorEphemeralKey(void)
+{
+    Crypto::P256SerializedKeypair serializedKeypairToCorrupt;
+    mEphemeralKey->Serialize(serializedKeypairToCorrupt);
+
+    *serializedKeypairToCorrupt.Bytes() ^= 0xFF;
+    ChipLogProgress(SecureChannel, "Fault Injected into Sigma3 InitiatorEphPubKey");
+
+    Crypto::P256Keypair * corruptedKeyPair;
+    corruptedKeyPair = mFabricsTable->AllocateEphemeralKeypairForCASE();
+    corruptedKeyPair->Deserialize(serializedKeypairToCorrupt);
```

Comment:
Might want to log (and bail) if this call fails.  And above if serialize fails.  Not all keystores support those...